src/asmcnc/comms/model_manager.py

```diff
@@ -38,25 +40,59 @@ def __init__(self, machine=None):
             self.machine = machine
             # we only need the update when we have no file yet.
             if not os.path.exists(self.PC_FILE_PATH):
-                self.machine.bind(setting_50=self.on_setting_50)
+                self.machine.s.bind(setting_50=self.on_setting_50)
 
         if self._initialized:
             return
         self._initialized = True
         # Do init here:
         self.set_machine_type(self.load_saved_product_code())
+        self._process_raw_migration_file()
+
+    def _process_raw_migration_file(self):
+        """This function will only run once when a new migration_raw.json file was added."""
+        if os.path.exists(self.MIGRATION_RAW_FILE_PATH):
+            with open(self.MIGRATION_RAW_FILE_PATH, 'r') as f:
+                d = json.load(f)
+                d['Pro Plus'] = [md5(s).hexdigest() for s in d['Pro Plus']]
+                d['Pro X'] = [md5(s).hexdigest() for s in d['Pro X']]
```

Comment:
Could you explain a bit about what's being done through this code? Just for my understanding.

Reply:
This could be a seperate helper script as well. It just converts the manually generated raw file and spices it up with some md5.